{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[6]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[6]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "one_loop_done\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[1]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[1]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[2]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[1]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[2]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[1]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[6]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "one_loop_done\n",
      "[[ 0.5263627   0.34873828  0.33896035  0.49389175 -0.52543247 -0.30898044\n",
      "  -0.10711589 -0.47052845 -0.1424828  -0.14937237  0.15975985 -0.10008753\n",
      "  -0.10358939 -0.27176213  0.28585932 -0.5132727 ]\n",
      " [-0.05564247 -0.5731126  -0.05833263  0.23979023  0.24939582 -0.17890437\n",
      "  -0.49066532 -0.37697408  0.00340888 -0.26836967  0.3054389  -0.1030006\n",
      "   0.19228025 -0.33447492  0.13123034 -0.70948327]]\n",
      "break\n",
      "[[ 0.5272573   0.35011956  0.33826324  0.4944174  -0.5248341  -0.30852917\n",
      "  -0.1074182  -0.47069523 -0.14199547 -0.15048088  0.15928327 -0.09984091\n",
      "  -0.10421174 -0.27147454  0.28570092 -0.5135449 ]\n",
      " [-0.05595401 -0.5728712  -0.05808366  0.23940754  0.24906968 -0.17877357\n",
      "  -0.48979416 -0.37693784  0.00309941 -0.26888236  0.30439627 -0.10187636\n",
      "   0.19248436 -0.3339682   0.1307147  -0.7095605 ]]\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[1]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[3]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[2]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[1]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[3]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[6]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[2]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "one_loop_done\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[2]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[6]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[2]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[0]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[5]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[4]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[6]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[3]\n",
      "boardscore\n",
      "Not finished Game\n",
      "[3]\n",
      "boardscore\n",
      "Not finished Game\n",
      "one_loop_done\n",
      "[[ 0.1775469   0.43356946 -0.49624678 -0.00159464 -0.2126632  -0.36580807\n",
      "  -0.3158147  -0.10726606 -0.03338629  0.37710276 -0.08823106 -0.29163566\n",
      "  -0.10447129 -0.27936473  0.04906451  0.53605604]\n",
      " [ 0.4708981  -0.3482476   0.198973    0.15016471  0.17929748 -0.20296915\n",
      "  -0.13506933  0.2743654   0.1541179   0.37638178  0.4978674   0.20420788\n",
      "  -0.3375988  -0.08923867  0.13549541  0.10818346]]\n",
      "break\n",
      "[[ 0.17778856  0.434004   -0.49561307 -0.00267737 -0.21330556 -0.36559004\n",
      "  -0.3152512  -0.1070394  -0.03393192  0.37875682 -0.0871447  -0.29151055\n",
      "  -0.10499594 -0.2786917   0.04893388  0.53672093]\n",
      " [ 0.47114226 -0.34824875  0.19885412  0.1514552   0.17792293 -0.2026113\n",
      "  -0.13484155  0.2748924   0.15457813  0.37553343  0.49898785  0.203643\n",
      "  -0.33670405 -0.08866213  0.13498965  0.10862781]]\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Parent directory of my_test_model doesn't exist, can't save.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFailedPreconditionError\u001b[0m                   Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1364\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1365\u001b[1;33m       \u001b[1;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1366\u001b[0m     \u001b[1;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mOpError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36m_run_fn\u001b[1;34m(feed_dict, fetch_list, target_list, options, run_metadata)\u001b[0m\n\u001b[0;32m   1349\u001b[0m       return self._call_tf_sessionrun(options, feed_dict, fetch_list,\n\u001b[1;32m-> 1350\u001b[1;33m                                       target_list, run_metadata)\n\u001b[0m\u001b[0;32m   1351\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36m_call_tf_sessionrun\u001b[1;34m(self, options, feed_dict, fetch_list, target_list, run_metadata)\u001b[0m\n\u001b[0;32m   1442\u001b[0m                                             \u001b[0mfetch_list\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtarget_list\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1443\u001b[1;33m                                             run_metadata)\n\u001b[0m\u001b[0;32m   1444\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mFailedPreconditionError\u001b[0m: Attempting to use uninitialized value conv2d/bias\n\t [[{{node save/SaveV2}}]]",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mFailedPreconditionError\u001b[0m                   Traceback (most recent call last)",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\u001b[0m in \u001b[0;36msave\u001b[1;34m(self, sess, save_path, global_step, latest_filename, meta_graph_suffix, write_meta_graph, write_state, strip_default_attrs, save_debug_info)\u001b[0m\n\u001b[0;32m   1175\u001b[0m               \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msaver_def\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msave_tensor_name\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1176\u001b[1;33m               {self.saver_def.filename_tensor_name: checkpoint_file})\n\u001b[0m\u001b[0;32m   1177\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36mrun\u001b[1;34m(self, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m    955\u001b[0m       result = self._run(None, fetches, feed_dict, options_ptr,\n\u001b[1;32m--> 956\u001b[1;33m                          run_metadata_ptr)\n\u001b[0m\u001b[0;32m    957\u001b[0m       \u001b[1;32mif\u001b[0m \u001b[0mrun_metadata\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36m_run\u001b[1;34m(self, handle, fetches, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1179\u001b[0m       results = self._do_run(handle, final_targets, final_fetches,\n\u001b[1;32m-> 1180\u001b[1;33m                              feed_dict_tensor, options, run_metadata)\n\u001b[0m\u001b[0;32m   1181\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_run\u001b[1;34m(self, handle, target_list, fetch_list, feed_dict, options, run_metadata)\u001b[0m\n\u001b[0;32m   1358\u001b[0m       return self._do_call(_run_fn, feeds, fetches, targets, options,\n\u001b[1;32m-> 1359\u001b[1;33m                            run_metadata)\n\u001b[0m\u001b[0;32m   1360\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\u001b[0m in \u001b[0;36m_do_call\u001b[1;34m(self, fn, *args)\u001b[0m\n\u001b[0;32m   1383\u001b[0m                     'disable_meta_optimizer = True')\n\u001b[1;32m-> 1384\u001b[1;33m       \u001b[1;32mraise\u001b[0m \u001b[0mtype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnode_def\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mop\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1385\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mFailedPreconditionError\u001b[0m: Attempting to use uninitialized value conv2d/bias\n\t [[node save/SaveV2 (defined at C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\framework\\ops.py:1751) ]]\n\nOriginal stack trace for 'save/SaveV2':\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\runpy.py\", line 193, in _run_module_as_main\n    \"__main__\", mod_spec)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py\", line 16, in <module>\n    app.launch_new_instance()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\traitlets\\config\\application.py\", line 664, in launch_instance\n    app.start()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel\\kernelapp.py\", line 563, in start\n    self.io_loop.start()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\platform\\asyncio.py\", line 148, in start\n    self.asyncio_loop.run_forever()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\asyncio\\base_events.py\", line 534, in run_forever\n    self._run_once()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\asyncio\\base_events.py\", line 1771, in _run_once\n    handle._run()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\asyncio\\events.py\", line 88, in _run\n    self._context.run(self._callback, *self._args)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\ioloop.py\", line 690, in <lambda>\n    lambda f: self._run_callback(functools.partial(callback, future))\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\ioloop.py\", line 743, in _run_callback\n    ret = callback()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 787, in inner\n    self.run()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 748, in run\n    yielded = self.gen.send(value)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 365, in process_one\n    yield gen.maybe_future(dispatch(*args))\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 209, in wrapper\n    yielded = next(result)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 272, in dispatch_shell\n    yield gen.maybe_future(handler(stream, idents, msg))\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 209, in wrapper\n    yielded = next(result)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel\\kernelbase.py\", line 542, in execute_request\n    user_expressions, allow_stdin,\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tornado\\gen.py\", line 209, in wrapper\n    yielded = next(result)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel\\ipkernel.py\", line 294, in do_execute\n    res = shell.run_cell(code, store_history=store_history, silent=silent)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\ipykernel\\zmqshell.py\", line 536, in run_cell\n    return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2855, in run_cell\n    raw_cell, store_history, silent, shell_futures)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2881, in _run_cell\n    return runner(coro)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n    coro.send(None)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3058, in run_cell_async\n    interactivity=interactivity, compiler=compiler, result=result)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3249, in run_ast_nodes\n    if (await self.run_code(code, result,  async_=asy)):\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3326, in run_code\n    exec(code_obj, self.user_global_ns, self.user_ns)\n  File \"<ipython-input-127-3c90e6f17eed>\", line 6, in <module>\n    saver = tf.train.Saver()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\", line 828, in __init__\n    self.build()\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\", line 840, in build\n    self._build(self._filename, build_save=True, build_restore=True)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\", line 878, in _build\n    build_restore=build_restore)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\", line 505, in _build_internal\n    save_tensor = self._AddSaveOps(filename_tensor, saveables)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\", line 206, in _AddSaveOps\n    save = self.save_op(filename_tensor, saveables)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\", line 122, in save_op\n    tensors)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\ops\\gen_io_ops.py\", line 1946, in save_v2\n    name=name)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\framework\\op_def_library.py\", line 793, in _apply_op_helper\n    op_def=op_def)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\util\\deprecation.py\", line 507, in new_func\n    return func(*args, **kwargs)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\framework\\ops.py\", line 3360, in create_op\n    attrs, op_def, compute_device)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\framework\\ops.py\", line 3429, in _create_op_internal\n    op_def=op_def)\n  File \"C:\\Users\\Phil\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\framework\\ops.py\", line 1751, in __init__\n    self._traceback = tf_stack.extract_stack()\n",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-127-3c90e6f17eed>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[0msess\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      6\u001b[0m \u001b[0msaver\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtrain\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSaver\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 7\u001b[1;33m \u001b[0msaver\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msave\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msess\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'my_test_model'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mglobal_step\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1000\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      8\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      9\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\tensorflow_core\\python\\training\\saver.py\u001b[0m in \u001b[0;36msave\u001b[1;34m(self, sess, save_path, global_step, latest_filename, meta_graph_suffix, write_meta_graph, write_state, strip_default_attrs, save_debug_info)\u001b[0m\n\u001b[0;32m   1191\u001b[0m               \"Parent directory of {} doesn't exist, can't save.\".format(\n\u001b[0;32m   1192\u001b[0m                   save_path))\n\u001b[1;32m-> 1193\u001b[1;33m         \u001b[1;32mraise\u001b[0m \u001b[0mexc\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1194\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1195\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mwrite_meta_graph\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Parent directory of my_test_model doesn't exist, can't save."
     ]
    }
   ],
   "source": [
    "%run REINFORCE.ipynb\n",
    "network_action = REINFORCEloop(epochs)\n",
    "\n",
    "\n",
    "sess = tf.Session()\n",
    "saver = tf.train.Saver()\n",
    "saver.save(sess, 'my_test_model',global_step=1000)\n",
    "\n",
    "\n",
    "#network_value = Valueloop(epochs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
